src/compiler/scala/tools/nsc/typechecker/Namers.scala

```diff
@@ -665,7 +665,7 @@ trait Namers extends MethodSynthesis {
       else assignAndEnterFinishedSymbol(tree)
 
       if (isEnumConstant(tree)) {
-        tree.symbol setInfo ConstantType(Constant(tree.symbol))
+        //tree.symbol setInfo ConstantType(Constant(tree.symbol))
```

Comment:
Looks like the field phase deletes all static fields under the mistaken assumption that all constant types don't need any storage...